src/action/AsyncUserActions.ts

```diff
@@ -460,3 +460,18 @@ export function loadManagedAssets(user: User) {
       });
   };
 }
+
+export function doesUsernameExists(username: string) {
+  const action = { type: ActionType.DOES_USERNAME_EXISTS };
+  return (dispatch: ThunkDispatch) => {
+    dispatch(asyncActionRequest(action, true));
+    return Ajax.get(
+      `${Constants.API_PREFIX}${USERS_ENDPOINT}/username`,
+      params({ username })
+    )
+      .then((data) =>
+        dispatch(asyncActionSuccessWithPayload(action, data === true))
```

Comment:
If you do not have a reducer for this action (and I don't see it anywhere), there is no point in calling `asyncActionSuccessWithPayload` and you can use just `asyncActionSuccess` and return the result.

Reply:
What about [result processing](https://github.com/kbss-cvut/termit-ui/blob/28f1b32e6877779648a1cb01507391f747204ff5/src/component/administration/user/CreateNewUserForm.tsx#L50)

I think it's better to have the possible promise result types "close together"  
`Promise<AsyncActionSuccess<boolean> | AsyncFailureAction>` looks more consistent than  
`Promise<boolean | AsyncFailureAction>`.
I can check for the actual value of `AsyncAction.status` with the former, and with the latter, I would check for status existence `AsyncAction.status === undefined`.

But maybe it's just personal preference. I can rewrite it.

Reply:
I don't like leaking async handling into components, that's the main reason. But I understand your reasoning w.r.t. close return types. I suppose the generic error handling in async actions is not as beneficial as I hoped it would be. Let's keep it this way then.